### ISSUE TITLE
Threaded messages have no thread indicator on the web if the mention is deleted

### ISSUE THREAD

**Issue by tyrosinase** [mastodon/mastodon#20964]
### Steps to reproduce the problem

1. Reply to godtributes
2. Tribby doesn't care about replies, so this is for everyone else - erase the mention.
3. ~~Post~~ **Publish!** the reply.

### Expected behaviour

Reply appears in public timelines with a "Show thread" link

### Actual behaviour

Reply appears in public timelines with no threading indication

### Detailed description

Replacing the "Show thread" with "Reply to $USER" is fine, except when it isn't a reply to that user. There needs to be some kind of indication that a reader can click on the post to see the thread.

### Specifications

Mastodon · v4.0.2

Firefox 106.0.5 Linux

**Comment by trwnh** [comment]
currently the only replies that show up in public timelines are self-replies (*not* "replies without a mention at the start")

you are still technically replying to a different user, even if you drop the mention; it just won't notify them

**Comment by tyrosinase** [comment]
You're right, what I was thinking of was "home timelines of non-mutual followers." Which I guess I might also be wrong about, but that's kind of a side note (unless it's changed since 3.x).

The actual bug/misfeature is: when it does show up wherever it does show up, _there is no longer a marker that it's a thread._

**Comment by simon-frankau** [comment]
I'd like to +1 the idea that a reply to a post that has been deleted should look more like a reply in the UI.

**Current situation:**

 * I make a reply.
 * Person I'm replying to deletes their post.
 * I view my reply, and it looks like a non-reply post, and I suddenly worry that I've somehow accidentally written a new post rather than a reply.

**Expected behaviour:**

 * I make a reply.
 * Person I'm replying to deletes their post.
 * I view my reply, and the previous message in the thread is a placeholder saying their post is unavailable (or whatever). I understand it's not my fault.

**Related issues:**

 * #11678 - This is about how you show the deleted post, not the reply, and IIUC requires tombstones.
 * #12264 / #7274 - These are about showing replies that still have an in-reply-to, just not visible.

**Suggested implementation:**

I think this can be handled as a relatively small, self-contained change, mostly (?) on the display side: In situations where the status is a reply, and the replied-to post would be displayed, but the `in_reply_to_id` field is null, instead display a "this post is unavailable" (or whatever the appropriate message is).
